pkg/kubelet/kubelet_test.go

```diff
@@ -2154,6 +2156,65 @@ func TestMakeEnvironmentVariables(t *testing.T) {
 				"KUBERNETES_RO_PORT_8087_TCP_ADDR=1.2.3.7"),
 			21,
 		},
+		{
+			"simple substitution",
+			&api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "unknown", Name: "test-pod"}},
+			&api.Container{
+				Env: []api.EnvVar{
+					{Name: "BAR", Value: "bar"},
+					{Name: "FOO", Value: "${BAR}"},
+				},
+			},
+			"kubernetes",
+			true,
+			util.NewStringSet(
+				"BAR=bar",
+				"FOO=bar",
+			),
+			2,
+		},
+		{
+			"slightly more complex substitution",
+			&api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "unknown", Name: "test-pod"}},
+			&api.Container{
+				Env: []api.EnvVar{
+					{Name: "BAR", Value: "bar"},
```

Comment:
Why bools for specific fields? I'm trying to avoid ad hoc approaches, which seem harder for users to learn and harder for us to maintain and extend.

How about just ObjectReference?

Speaking of which, we should change its field syntax to be more standard.